Woof/Woof/ViewModel/Owner/OwnerWalkingsViewModel.swift

```diff
@@ -27,9 +27,19 @@ final class OwnerWalkingsViewModel: ObservableObject {
 
     private var walkings: [Walking] = [] {
         didSet {
-            futureWalkings = walkings.filter { $0.start >= Date.now }
-            currentWalkings = walkings.filter { $0.start <= Date.now && $0.end > Date.now }
-            finishedWalkings = walkings.filter { $0.end <= Date.now }
+            futureWalkings = []
+            currentWalkings = []
+            finishedWalkings = []
+
+            for walking in walkings {
+                if walking.start >= .now {
+                    futureWalkings.append(walking)
+                } else if walking.start <= .now && walking.end > .now {
+                    currentWalkings.append(walking)
+                } else {
+                    finishedWalkings.append(walking)
+                }
```

Comment:
It might be a bit efficient. We don't trigger redrawing sections if they don't change.   It's just an example to get to know with drawing efficiency, it's not crucial )
 WDYT?
```suggestion
            var newFutureWalkings = [Walking]()
            var newCurrentWalkings = [Walking]()
            var newFinishedWalkings = [Walking]()
            
            for walking in walkings {
                if walking.start >= .now {
                    newFutureWalkings.append(walking)
                } else if walking.start <= .now && walking.end > .now {
                    newCurrentWalkings.append(walking)
                } else {
                    newFinishedWalkings.append(walking)
                }
            }
            
            if newFutureWalkings != futureWalkings {
                futureWalkings = newFutureWalkings
            }
            if newCurrentWalkings != currentWalkings {
                currentWalkings = newCurrentWalkings
            }
            if newFinishedWalkings != finishedWalkings {
                finishedWalkings = newFinishedWalkings
            }
```
```suggestion
            futureWalkings = []
            currentWalkings = []
            finishedWalkings = []

            for walking in walkings {
                if walking.start >= .now {
                    futureWalkings.append(walking)
                } else if walking.start <= .now && walking.end > .now {
                    currentWalkings.append(walking)
                } else {
                    finishedWalkings.append(walking)
                }
```